### PR TITLE
Task/116 create utxo wallet

### PR DESCRIPTION
**Pull request by storkandstars**
# Overview

Initilizing sio_core library with generated mnemonic. Mnemonic is encrypted by default and decrepted after verifying pin. User can fetch available crypto assets from our API and filter them by `name` and `ticker`. Assets are grouped by asset type that is defined by `assetId` (Asset ID is internally defined numeric value representing and any type of asset). User can add a specific network wallet (An asset that lives on a specific blockchain network).

## Test cases:

1. Start the app and login. You should be prompt to set 4 digit pin.
2. After setting and submitting newly set pin, you should be redirected to a success page.
3. click on `go to app` button.
4. After setting pin there should not be any pin verification however if application is refreshed you will be asked to insert a pin
5. Go to inventory and click on search. There should be a loading animation and then assets should be listed.
6. Open and asset and its networks should be showed in dropdown.
7. Click on plus button. Note - there is no filtering at this moment.
8. Close the search page and you should se a newly created asset wallet.
9. Close the app and enter it again. Asset wallet should be persistent
10. Go to search and filter an asset by name or ticker. The list should be filtered.

### CODE REVIEW

File: lib/view/routes/guards/protected_guard.dart [hunk 1]
```diff
@@ -123,9 +130,7 @@ class _Protection extends StatelessWidget {
                           attempts > 1
                               ? "${context.locale.protectedErrorRemaining} $attempts"
                               : context.locale.protectedErrorLast,
-                          style: TextStyle(
-                            color: Theme.of(context).colorScheme.error,
-                          ),
+                          style: const TextStyle(color: Colors.red),
```
Comment (jvrastil) [inline lib/view/routes/guards/protected_guard.dart:133]: why did you revert this change?

File: lib/view/screens/application_screen.dart [hunk 2]
```diff
@@ -55,6 +53,7 @@ class _ApplicationScreenState extends State<ApplicationScreen>
                           onTap: (context, key) {
                             GoRouter.of(context)
                                 .goNamed(AuthenticatedRouter.dashboard);
+                            // ApplicationRoute.key.currentState
```
Comment (jvrastil) [inline lib/view/screens/application_screen.dart:56]: is this supposed to stay here?

File: lib/view/widgets/asset_wallet_expansion_list.dart [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        (a) {
+          final asset = Assets.getAsset(a.assetId);
+          return ExpansionPanelRadio(
+            backgroundColor: Theme.of(context).scaffoldBackgroundColor,
+            value: UniqueKey(),
+            canTapOnHeader: true,
+            headerBuilder: (context, isExpanded) {
+              return Padding(
+                padding: const EdgeInsets.symmetric(
+                  vertical: 20.0,
```
Comment (jvrastil) [inline lib/view/widgets/asset_wallet_expansion_list.dart:27]: there is `CommonTheme.verticalPadding`

File: lib/view/widgets/crypto_asset_expansion_list.dart [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+          final asset = Assets.getAsset(a.assetId);
+
+          return ExpansionPanelRadio(
+            backgroundColor: Theme.of(context).scaffoldBackgroundColor,
+            value: UniqueKey(),
+            canTapOnHeader: true,
+            headerBuilder: (context, _) {
+              return Padding(
+                padding:
+                    const EdgeInsets.symmetric(vertical: 20.0, horizontal: 0),
```
Comment (jvrastil) [inline lib/view/widgets/crypto_asset_expansion_list.dart:30]: same here

File: lib/view/widgets/pin_digits.dart [hunk 5]
```diff
@@ -15,7 +15,7 @@ class PinDigits extends StatefulWidget {
     this.pin,
     this.pinLength = ValidatedPin.length,
     this.pinDigitStyle = PinDigitStyle.hideAllAfterTime,
-    this.duration = const Duration(milliseconds: 1000),
+    this.duration = const Duration(milliseconds: 0),
```
Comment (jvrastil) [inline lib/view/widgets/pin_digits.dart:18]: probably missed value from rebase

File: lib/view/widgets/wallet_list_item.dart [hunk 6]
```diff
@@ -22,21 +22,11 @@ class _WalletListItem extends State<WalletListItem> {
           padding: CommonTheme.padding,
```
Comment (jvrastil) [inline lib/view/widgets/wallet_list_item.dart:22]: is this widget still needed or can we remove it?

**Review by jvrastil — CHANGES_REQUESTED** [review]

File: lib/view/widgets/pin_digits.dart (hunk 5 above)
Comment (storkandstars) [reply lib/view/widgets/pin_digits.dart]: Yes. It was bad previous rebase. Fix [97795b0](https://github.com/SimplioOfficial/simplio-app/pull/128/commits/97795b0ec18534415c10599e69f94646d83dcb36)

File: lib/view/widgets/asset_wallet_expansion_list.dart (hunk 3 above)
Comment (storkandstars) [reply lib/view/widgets/asset_wallet_expansion_list.dart]: Fixed [f6a1f8e](https://github.com/SimplioOfficial/simplio-app/pull/128/commits/f6a1f8e00c42b926e4ac43e68e7029bd5a0f360f)

File: lib/view/widgets/crypto_asset_expansion_list.dart (hunk 4 above)
Comment (storkandstars) [reply lib/view/widgets/crypto_asset_expansion_list.dart]: Fixed [f6a1f8e](https://github.com/SimplioOfficial/simplio-app/pull/128/commits/f6a1f8e00c42b926e4ac43e68e7029bd5a0f360f)

File: lib/view/screens/application_screen.dart (hunk 2 above)
Comment (storkandstars) [reply lib/view/screens/application_screen.dart]: Fixed [83373da](https://github.com/SimplioOfficial/simplio-app/pull/128/commits/83373da74b02c12dad927705d72adadfece146f1)

File: lib/view/routes/guards/protected_guard.dart (hunk 1 above)
Comment (storkandstars) [reply lib/view/routes/guards/protected_guard.dart]: Fixed [9681d8c](https://github.com/SimplioOfficial/simplio-app/pull/128/commits/9681d8c233b92c1b6294ceeabcfc2c2ca9f5fc6f)

File: lib/data/model/account_wallet.dart [hunk 7]
```diff
@@ -1,114 +1,138 @@
...
+    required bool isBackedUp,
+    required bool isImported,
+  }) : this(
+          value: LockableString.from(base64String: base64Mnemonic),
+          isBackedUp: isBackedUp,
+          isImported: isImported,
+        );
 
-  LockableSeed.from({
+  LockableMnemonic.value({
```
[the remark this answers is not in the document]
Comment (storkandstars) [reply lib/data/model/account_wallet.dart]: Updated here [050d4a0](https://github.com/SimplioOfficial/simplio-app/pull/128/commits/050d4a0b70df5cced1ef820d92f0183a643d92c2)

File: lib/data/repositories/auth_repository.dart [hunk 8]
```diff
@@ -65,58 +58,59 @@ class AuthRepository with JwtMixin {
     throw Exception("Sign up has failed");
   }
 
+  Future<Account> _resolveAccount(String accountId) async {
```
[the remark this answers is not in the document]
Comment (storkandstars) [reply lib/data/repositories/auth_repository.dart]: Updated here [8778695](https://github.com/SimplioOfficial/simplio-app/pull/128/commits/8778695752924487286e716c938be933371188c7)

File: .github/workflows/_test_application.yml [hunk 9]
```diff
@@ -64,7 +64,7 @@ jobs:
       - uses: actions/download-artifact@v3
         with:
           name: generated-files-and-packages
-      - run: flutter test test/unit_tests
+      - run: flutter test
```
[the remark this answers is not in the document]
Comment (storkandstars) [reply .github/workflows/_test_application.yml:67]: We disgusted that we will return to previous test folder structure. Due to best practices.

File: lib/view/widgets/wallet_list_item.dart (hunk 6 above)
Comment (storkandstars) [reply lib/view/widgets/wallet_list_item.dart]: Removed [91a93fa](https://github.com/SimplioOfficial/simplio-app/pull/128/commits/91a93fae236157dc9c298a98a67eab3dbaebdb9a)

File: .github/workflows/_test_application.yml (hunk 9 above)
[the remark this answers is not in the document]
Comment (jvrastil) [reply .github/workflows/_test_application.yml:67]: ok, in that case you need to change `    uses: SimplioOfficial/simplio-app/.github/workflows/_test_application.yml@test/fix_long_execution` into `    uses: SimplioOfficial/simplio-app/.github/workflows/_test_application.yml@task/116-create-utxo-wallet` in file `.github/workflows/run_tests_on_pull_request.yml`

[the remark this answers is not in the document]
Comment (storkandstars) [reply .github/workflows/_test_application.yml:67]: Resolved b52d60e

**Review by jvrastil — APPROVED** [review]
looks good now 👍

**Pull request merged**